### ISSUE TITLE
Performance Drop Due to Padding Changes or Refactor?

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#256]
I re-ran a benchmark script for a Tensorflow model that I'd last run on September 5th, and found a significant drop in R^2 in the trained model. It's not clear to me what the source of the performance drop is. There have been some major changes since September 5th:
- #233 performed a large-scale refactoring of the `Model` class, and made large changes to the `TensorflowModel` classes.
- #240 made changes to support python 3.
- #244 made changes to support prediction for models where batch-size was set rigidly.

Any of these changes could potentially have introduced the performance bug (part of reason why #255 becomes quite important). There are a couple options for debugging moving forward. One would be to look back in the git history and use the version of DeepChem from September 5th to run the benchmark script to verify that the breaking change was introduced recently. Another option is to keep running experiments to try to understand the source of the performance difference.

**Comment by rbharath** [comment]
I believe the underlying bug was caught and fixed in #289. Going to go ahead and close.
